File: src/main/java/de/burrotinto/burroPlayer/interfaces/serial/executors/MVCRandomExecutor.java
package de.burrotinto.burroPlayer.interfaces.serial.executors;

import de.burrotinto.burroPlayer.adapter.random.RandomAdapter;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

/**
 * Created by derduke on 08.03.17.
 */
@Service
@RequiredArgsConstructor
public class MVCRandomExecutor implements RandomExecutor {

    private final RandomAdapter randomAdapter;

    @Override
    public void execute(int command) {
        randomAdapter.playRandom();
    }
}

File: src/main/kotlin/de/burrotinto/burroPlayer/interfaces/serial/executors/MVCRandomExecutor.kt
package de.burrotinto.burroPlayer.interfaces.serial.executors

import de.burrotinto.burroPlayer.adapter.random.RandomAdapter
import org.springframework.stereotype.Service

/**
 * Created by derduke on 08.03.17.
 */
@Service
class MVCRandomExecutor(private val randomAdapter: RandomAdapter) : RandomExecutor {

    override fun execute(command: Int) {
        randomAdapter.playRandom()
    }
}
